dawn/src/dawn/Optimizer/PassSetStageLocationType.h

```diff
@@ -0,0 +1,47 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#pragma once
+
+#include "Pass.h"
+
+namespace dawn {
+
+/// @brief
+/// * Input: any IIR where GridType == Unstructured and where statements inside the same stage are
```

Comment:
it actually also requires every statement to be in it's own stage, see the assert in line 91

Reply:
@Stagno What do you think, should we make this comment more restrictive or delete the assert?
@twicki The reason for phrasing it like this was that strictly, the single statement per stage is not required (except for the assert) as long as each stage has only statements on the same location type.

Reply:
I'm for deleting the assert

Reply:
deleted the assert